src/pigui/PerfInfo.cpp

```diff
@@ -405,6 +407,32 @@ void PerfInfo::DrawWorldViewStats()
 			if (system)
 				system->Dump(Log::GetLog()->GetLogFileHandle());
 		}
+
+		if (ImGui::Button("Dump System-Editor JSON")) {
+			SystemPath path = Pi::game->GetSectorView()->GetSelected();
+			RefCountedPtr<StarSystem> system = Pi::game->GetGalaxy()->GetStarSystem(path);
+
+			if (system) {
+				char fileName[1024];
+				sprintf(fileName, "%s.json", system->GetName().c_str());
```

Comment:
I'd recommend using `fmt::format("{}.json", system->GetName())` or `system->GetName() + ".json"` - this avoids stack-allocating a 1kb buffer to store a grand total of ~32 bytes of string data. It *also* avoids the improbable but possible chance of overflowing the buffer (given that you're not using `snprintf`). The heap allocation is completely negligible in this case, especially given that it's being further processed through `FileSystem::JoinPathBelow`.

Reply:
Yeah you're right, that was me being extremely lazy :D